### ISSUE TITLE
ReflectionException: Class checkForMaintenanceMode does not exist

### ISSUE THREAD

**Issue by DanielFurmanov** [cnvs/easel#119]
I am updated from Laravel 5.2 to Laravel 5.4 and installed Canvas. 

**TL;DR** Laravel could not instantiate Reflection using blog.php settings (middleware part of them) 

So here is a screenshot of a problem: ![reflection error](http://dl4.joxi.net/drive/2017/04/04/0018/2806/1227510/10/32fee954a9.jpg)

That happens when I am trying to go to `/blog` canvas registered route.
The troubling line is [config/blog.php:125](https://github.com/cnvs/easel/blob/master/config/blog.php#L125) and [105](https://github.com/cnvs/easel/blob/master/config/blog.php#L105)

If I replace  `'checkForMaintenanceMode'` with `\Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode::class` in this file, I am starting to get another `ReflectionException`  "Class canvasInstalled does not exist"

Somehow `../vendor/composer/ClassLoader.php:329` could not parse the file from the string from settings
(xdebug):
![xdebug](http://dl3.joxi.net/drive/2017/04/04/0018/2806/1227510/10/f75dfc4821.jpg)

even if I replace `'canvasInstalled'` with `Canvas\Http\Middleware\EnsureInstalled`

*Any ideas what causes this and what could be done?*

**Update:** Also noticed that `php artisan route:list` throws
````
[ErrorException]       
  mkdir(): Invalid path

**Comment by austintoddj** [comment]
We fully intend to add support for Laravel 5.4. Currently, the hold up is with `proengsoft/laravel-jsvalidation`. They have an open pull request at [proengsoft/laravel-jsvalidation#224](https://github.com/proengsoft/laravel-jsvalidation/pull/224)that provides support for Laravel 5.4. Until I see that merged and closed, we can't move forward in upgrading Canvas, unless we choose to go with another validator.

So your issue will definitely be addressed when we upgrade. We are just waiting on Proengsoft to merge this PR and we can get moving on it.

**Comment by khaledelmahdi** [comment]
Hi,

The pull request has been merged now.

Can anyone help please?

Thanks.

**Comment by austintoddj** [comment]
Just saw that PR get merged in. Looks like there may still be [an outstanding issue](https://github.com/proengsoft/laravel-jsvalidation/pull/224), but I'm keeping a close eye on this and will move this forward as soon as I see a solid resolution there.

**Comment by austintoddj** [comment]
@DanielFurmanov @khaledelmahdi We've taken a different route now and removed the JSValidation package all together. As soon as I iron out a couple more issues, I will begin the upgrade to L5.4.

**Comment by austintoddj** [comment]
@DanielFurmanov Found the bug here. Looks like a somewhat obscure change was made [here](https://laravel.com/docs/5.4/upgrade) pertaining to the `middleware` method. After updating the Service Provider ([b8328bc](https://github.com/cnvs/easel/commit/b8328bc6331d4ccc3bfb6110a855bf1e48f78221)), the middleware loads properly.

Continuing on with the Laravel 5.4 upgrade.
